plugins/base/src/main/kotlin/parsers/MarkdownParser.kt

```diff
@@ -10,503 +10,480 @@ import org.intellij.markdown.ast.impl.ListItemCompositeNode
-import org.jetbrains.kotlin.idea.kdoc.resolveKDocLink
 import org.jetbrains.kotlin.kdoc.parser.KDocKnownTag
 import org.jetbrains.kotlin.kdoc.psi.impl.KDocSection
 import org.jetbrains.kotlin.kdoc.psi.impl.KDocTag
 import java.net.MalformedURLException
 import java.net.URL
 import org.intellij.markdown.parser.MarkdownParser as IntellijMarkdownParser
 
-class MarkdownParser(
-    private val resolutionFacade: DokkaResolutionFacade? = null,
-    private val declarationDescriptor: DeclarationDescriptor? = null,
-    private val logger: DokkaLogger
+open class MarkdownParser(
+    private val externalDri: (String) -> DRI?
 ) : Parser() {
 
-    inner class MarkdownVisitor(val text: String, val destinationLinksMap: Map<String, String>) {
-
-        private fun headersHandler(node: ASTNode): DocTag =
-            DocTagsFromIElementFactory.getInstance(
-                node.type,
-                visitNode(node.children.find { it.type == MarkdownTokenTypes.ATX_CONTENT }
-                    ?: throw IllegalStateException("Wrong AST Tree. ATX Header does not contain expected content")).children
-            )
-
-        private fun horizontalRulesHandler(node: ASTNode): DocTag =
-            DocTagsFromIElementFactory.getInstance(MarkdownTokenTypes.HORIZONTAL_RULE)
-
-        private fun emphasisHandler(node: ASTNode): DocTag =
-            DocTagsFromIElementFactory.getInstance(
-                node.type,
-                children = node.children.evaluateChildrenWithDroppedEnclosingTokens(1)
-            )
+    override fun parseStringToDocNode(extractedString: String): DocTag {
+        val gfmFlavourDescriptor = GFMFlavourDescriptor()
+        val markdownAstRoot: ASTNode =
+            IntellijMarkdownParser(gfmFlavourDescriptor).buildMarkdownTreeFromString(extractedString)
+        destinationLinksMap = getAllDestinationLinks(extractedString, markdownAstRoot).toMap()
+        text = extractedString
+        return visitNode(markdownAstRoot)
+    }
 
-        private fun strongHandler(node: ASTNode): DocTag =
-            DocTagsFromIElementFactory.getInstance(
-                node.type,
-                children = node.children.evaluateChildrenWithDroppedEnclosingTokens(2)
-            )
+    override fun preparse(text: String) = text
 
-        private fun List<ASTNode>.evaluateChildrenWithDroppedEnclosingTokens(count: Int) =
-            drop(count).dropLast(count).evaluateChildren()
+    private lateinit var destinationLinksMap: Map<String, String>
+    private lateinit var text: String
 
-        private fun blockquotesHandler(node: ASTNode): DocTag =
-            DocTagsFromIElementFactory.getInstance(
-                node.type, children = node.children
-                    .filterIsInstance<CompositeASTNode>()
-                    .evaluateChildren()
-            )
+    override fun parse(text: String): DocumentationNode {
+        return DocumentationNode(listOf(Description(parseStringToDocNode(text))))
+    }
```

Comment:
You should not override default behaviour from base class, especially not in that way

Reply:
Just delete it, so the base class `parse` will be dispatched